state/user.go

```diff
@@ -543,6 +540,61 @@ func (u *User) ensureNotDeleted() error {
 	return nil
 }
 
+// ResetPassword cleans up password related field.
+// It generates and returns a new user secret key.
+// This must be an active user.
+func (u *User) ResetPassword() ([]byte, error) {
+	var key []byte
+	buildTxn := func(attempt int) ([]txn.Op, error) {
+		if err := u.ensureNotDeleted(); err != nil {
+			return nil, errors.Trace(err)
+		}
+		if u.IsDisabled() {
+			return nil, fmt.Errorf("user deactivated")
```

Comment:
I am not convinced that this needs further clarification. The overall error that will come from here will be `cannot reset password for user "bob": user deactivated`.